portfolio/src/main/webapp/script.js

```diff
@@ -50,6 +50,43 @@ async function getMessages(){
   })
 }
 
+/**
+ * Displays the comment form if the user is logged in alongside
+ * a logout link. Displays a login link if the user is not loggedin.
+ */
+async function unhideCommentForm(){
```

Comment:
This function doesn't always unhide the comment form, right? Sometimes it keeps the comment form hidden and displays a login link? If so, maybe change this to something like `showCommentFormOrLoginLink()` to make it more obvious exactly what it's doing?